TicTacToe.cpp

```diff
@@ -103,3 +103,29 @@ void TicTacToe::viewAll()
 	std::cout << "|/     /     /   " << std::endl;
 	std::cout << field[0][0][0] << " --- " << field[0][1][0] << " --- " << field[0][2][0] << "    " << std::endl;
 }
+
+void TicTacToe::viewTop() {
+ 	   for (int y = 2; y >= 0; y--) {
+        for (int x = 0; x < 3; x++) {
+            std::deque<int> square = field[y][x];
```

Comment:
There isn't much of a performance impact here because our `deque`s are at most three elements long, but it isn't generally good practice to perform copy operations like this. Essentially, what happens when you create a new `deque` object and assign it to be an equal to another one is that the `field[i][j]` object is copied in its entirety and duplicated into the new `square` object.

This can be decently computationally expensive, so if you wanted to do this to simplify the code down below, like why I believe you write this, you'd create a pointer instead. Pointers don't do the same copy operation because they just point to something else, and so the copy operation's time cost is avoided.